clients/vault/src/system.rs

```diff
@@ -150,11 +156,32 @@ impl VaultIdManager {
 		self.vault_data.read().await.get(vault_id).map(|x| x.stellar_wallet.clone())
 	}
 
+	pub async fn get_active_vault(&self, vault_id: &VaultId) -> Option<VaultData> {
+		let vault = self.vault_data.read().await.get(vault_id)?.clone();
+		// Filter liquidated
+		if vault.liquidated {
+			return None;
+		}
+		return Some(vault);
+	}
+
 	pub async fn get_vault(&self, vault_id: &VaultId) -> Option<VaultData> {
 		self.vault_data.read().await.get(vault_id).cloned()
 	}
 
+	// Get all ACTIVE vaults
```

Comment:
I didn't want to change the name of this method to `get_active_entries` since it is used in other places and wanted to minimize the modifications to the rest of the code, but we could of course.